src/backend/api/meals.js

```diff
@@ -2,14 +2,199 @@ const express = require("express");
+
+router.get("/", async(request, response) => {
+
+    try {
+        // knex syntax for selecting things. Look up the documentation for knex for further info
+        const meals = await knex("meals");
+
+        if (request.query.hasOwnProperty('id')) {
+
+            const requestId = Number(request.query.id);
+            const requestedMeal = await knex('meals')
+                .select()
+                .where('id', '=', requestId);
+
+            console.log(requestedMeal);
+            console.log(typeof(requestedMeal));
+            if (Object.keys((requestedMeal)).length !== 0) {
+
+                response.json(requestedMeal);
+            } else {
+                response.status(404).json(`No meal matches the id: ${requestId}`);
+            }
+
+        } else if (Object.keys(request.query).length > 1 &&
+            request.query.hasOwnProperty('maxPrice', 'limit')) {
+
+            const requestPrice = Number(request.query.maxPrice);
+            const requestLimit = Number(request.query.limit);
+            const mealFinder = meals.some(meal => meal.price <= requestPrice);
+
+            if (requestLimit !== 0 && mealFinder) {
+
+                const requestOrder = await knex('meals').select()
+                    .where('price', '<=', requestPrice)
+                    .limit(requestLimit)
+                    .orderBy('price');
+                response.json(requestOrder);
+            } else {
+                response.status(404).json(`NO meal is cheaper than ${requestPrice}`);
+            }
+
+        } else if (request.query.hasOwnProperty('maxPrice')) {
+
+            const requestPrice = Number(request.query.maxPrice);
+            const mealFinder = meals.some(meal => meal.price <= requestPrice);
+            const requestedMeal = await knex('meals')
+                .select()
+                .where('price', '<=', requestPrice);
+
+            if (Object.keys((requestedMeal)).length !== 0) {
+                response.json(requestedMeal);
+            } else {
+                response.status(404).json(`NO meal is cheaper than ${requestPrice}`);
+            }
+
+        } else if (request.query.hasOwnProperty('createdAfter')) {
+
+            const requestDate = new Date(request.query.createdAfter);
+            const mealFinder = meals.some(meal => meal.created_date > requestDate);
+            const requestedMeal = await knex('meals')
+                .select()
+                .where('created_date', '>', requestDate);
+
+            if (Object.keys((requestedMeal)).length !== 0) {
+                response.json(requestedMeal);
+            } else {
+                response.status(404).json(`NO meal has been created after: ${requestDate}`);
+            }
+
+        } else if (request.query.hasOwnProperty('limit')) {
+            const requestLimit = Number(request.query.limit);
+            const requestedLimit = await knex('meals')
+                .select()
+                .limit(requestLimit);
+
```

Comment:
Perhaps would make sense to move `requestedLimit` into the conditional check below? 
So you don't query before you know if `requestLimit` is not 0.